src/fsharp/FSharp.Core/prim-types.fs

```diff
@@ -4509,7 +4509,10 @@ namespace Microsoft.FSharp.Core
         [<CompiledName("ToString")>]
         let inline string (value: 'T) = 
              anyToString "" value
-             when 'T : string     = (# "" value : string #)     // force no-op
+
+             when 'T : string =
+                if value = unsafeDefault<'T> then ""
+                else (# "" value : string #)     // force no-op
```

Comment:
The fix is needed for these lines too (further down below)

```
             when 'T : StringBuilder = let x = (# "" value : StringBuilder #) in x.ToString()
             when 'T : IFormattable = let x = (# "" value : IFormattable #) in x.ToString(null, CultureInfo.InvariantCulture)
```

Reply:
@dsyme good eyes, on it :-)